packages/kit/src/pages.ts

```diff
@@ -11,3 +13,28 @@ export function extendPages (cb: NuxtHooks['pages:extend']) {
     nuxt.hook('pages:extend', cb)
   }
 }
+
+export interface ExtendRouteRulesOptions {
+  /**
+   * Override route rule config
+   *
+   * @default false
+   */
+  override?: boolean
+}
+
+export function extendRouteRules (route: string, rule: NitroRouteConfig, options: ExtendRouteRulesOptions = {}) {
+  const nuxt = useNuxt()
+  if (!nuxt.options.routeRules) {
+    nuxt.options.routeRules = {}
+  }
+  nuxt.options.routeRules[route] = options.override
+    ? rule
+    : defu(rule, nuxt.options.routeRules[route] || {})
```

Comment:
I think this is what @pi0 is suggesting:

```suggestion
  nuxt.options.routeRules[route] = options.override
    ? defu(rule, nuxt.options.routeRules[route])
    : defu(nuxt.options.routeRules[route], rule)
```

Reply:
I thing in this example, override doesn't mean.
I thing remove override option is better! Or I misunderstood :(